runtime/src/invoke_dpe.rs

```diff
@@ -23,21 +27,44 @@ impl InvokeDpeCmd {
             let hashed_rt_pub_key = crypto
                 .hash(AlgLen::Bit384, &rt_pub_key.to_der()[1..])
                 .map_err(|_| CaliptraError::RUNTIME_INITIALIZE_DPE_FAILED)?;
+            let image_header = pdata.manifest1.header;
             let mut env = DpeEnv::<CptraDpeTypes> {
                 crypto,
                 platform: DpePlatform::new(
-                    pdata.manifest1.header.pl0_pauser,
+                    image_header.pl0_pauser,
                     hashed_rt_pub_key,
                     &mut drivers.cert_chain,
                 ),
             };
 
-            match drivers
-                .persistent_data
-                .get_mut()
-                .dpe
-                .execute_serialized_command(&mut env, drivers.mbox.user(), &cmd.data)
-            {
+            let locality = drivers.mbox.user();
+            let command = Command::deserialize(&cmd.data)
+                .map_err(|_| CaliptraError::RUNTIME_INVOKE_DPE_FAILED)?;
+            let mut dpe = &mut drivers.persistent_data.get_mut().dpe;
+            let resp = match command {
+                Command::GetProfile => Ok(Response::GetProfile(
+                    dpe.get_profile(&mut env.platform)
+                        .map_err(|_| CaliptraError::RUNTIME_INVOKE_DPE_FAILED)?,
+                )),
+                Command::InitCtx(cmd) => cmd.execute(dpe, &mut env, locality),
+                Command::DeriveChild(cmd) => cmd.execute(dpe, &mut env, locality),
+                Command::CertifyKey(cmd) => {
+                    // PL1 cannot request X509
+                    if cmd.format == CertifyKeyCmd::FORMAT_X509 && image_header.flags & 1 == 0 {
```

Comment:
I think this misses checking whether locality == pl0_pauser. I also think checking flags directly might be a bit confusing. Maybe a helper like this would be good:

```rust
fn is_caller_pl0(drivers: &mut Drivers, caller_locality: u32) -> bool
```

Reply:
Done